pytorch_blade/torch_blade/quantization/prepare_data.py

```diff
@@ -0,0 +1,247 @@
+
+            # Combine the inputs of the Fusion Group into a list
+            # and make the observer record the list
+            forward_node = users[0]
+            list_pack = self.graph.create('prim::ListConstruct')
+            self.graph.appendNode(list_pack)
+            list_pack.moveBefore(forward_node)
+            for inp in forward_node.input_list()[1:]:
+                list_pack.addInput(inp)
+            list_pack.output().setType(torch_blade.tools.get_list_tensor_type())
+
+            # insert the observer, three steps:
+            #   1. register the observer
+            #   2. create getattr node for the observer
+            #   3. create call method node for the observer
+            collect_observer = DataCollectObserver()
+            collect_observer = torch.jit.script(collect_observer)
+            observer_name = f"{self.collect_observer_prefix}_{idx}"
+            all_collect_observer_name.append(observer_name)
+            self.custom_module_owner._c._register_attribute(
+                observer_name,
+                collect_observer._c._type(),
+                collect_observer._c
+            )
+            get_attr = self.graph.createGetAttr(self.custom_module_owner_inp, observer_name)
+            get_attr.output().setType(collect_observer._c._type())
+            self.graph.appendNode(get_attr)
+            get_attr.moveAfter(list_pack)
+
+            call_method = self.graph.create("prim::CallMethod")
+            self.graph.appendNode(call_method)
+            call_method.output().setType(torch_blade.tools.get_list_tensor_type())
+            call_method.s_("name", "forward")
+            call_method.addInput(get_attr.output())
+            call_method.addInput(list_pack.output())
+            call_method.moveAfter(get_attr)
+
+            # unpack the output of the forward method
+            list_unpack = self.graph.create('prim::ListUnpack')
+            self.graph.appendNode(list_unpack)
+            list_unpack.moveAfter(call_method)
+            list_unpack.addInput(call_method.output())
+            list_unpack.eraseOutput(0)
+            for inp in list_pack.input_list():
+                lu_output = list_unpack.addOutput()
+                lu_output.setType(inp.type())
+                inp.replaceAllUsesAfterNodeWith(list_unpack, lu_output)
+        self.all_collect_observer_name = all_collect_observer_name
+
+    def get_calib_data_for_each_group(self):
+        all_data = []
+        self.c_module.create_method_from_graph("inference", self.graph)
+        for data in self.calibration_data:
+            self.c_module.inference(*data, self.custom_module_owner)
+
+        for observer_name in self.all_collect_observer_name:
+            data = self.custom_module_owner.__getattr__(observer_name).data
+            # the first one is dummy data
+            all_data.append(data[1:])
+        return all_data
+
+    def save_data(self, data):
+        tmp_dir = tempfile.mkdtemp(prefix="torch_blade")
```

Comment:
Will this tmp file be removed after the DataPreparer is released?

Reply:
It's better to remove the temp file after it's used.

Reply:
Yes, while the trt ptq is enabled, codes about deleting these temp files will be added.